File: unity-plugin/src/main/java/com/example/blehid/unity/events/AdvertisingStateChangedEvent.java
package com.example.blehid.unity.events;

/**
 * Event fired when the advertising state changes.
 */
public class AdvertisingStateChangedEvent extends BaseEvent {
    private final boolean isAdvertising;
    
    /**
     * Creates a new advertising state changed event.
     * 
     * @param isAdvertising Whether the device is currently advertising
     */
    public AdvertisingStateChangedEvent(boolean isAdvertising) {
        super(EventType.ADVERTISING_STATE_CHANGED);
        this.isAdvertising = isAdvertising;
    }
    
    /**
     * Gets whether the device is currently advertising.
     * 
     * @return True if advertising, false otherwise
     */
    public boolean isAdvertising() {
        return isAdvertising;
    }
    
    @Override
    public String toString() {
        return String.format("%s[isAdvertising=%b]", super.toString(), isAdvertising);
    }
}

File: unity-test/Assets/BleHidTest/Events/AdvertisingStateChangedEvent.cs
using System;

namespace BleHid.Events
{
    /// <summary>
    /// Event fired when the advertising state changes.
    /// </summary>
    public class AdvertisingStateChangedEvent : BleHidEvent
    {
        /// <summary>
        /// Whether the device is currently advertising
        /// </summary>
        public bool IsAdvertising { get; private set; }
        
        /// <summary>
        /// Creates a new advertising state changed event.
        /// </summary>
        /// <param name="isAdvertising">Whether the device is currently advertising</param>
        /// <param name="eventId">ID of the event</param>
        /// <param name="javaTimestamp">Java timestamp (milliseconds since epoch)</param>
        public AdvertisingStateChangedEvent(bool isAdvertising, Guid eventId, long javaTimestamp)
            : base(BleHidEventType.AdvertisingStateChanged, eventId, javaTimestamp)
        {
            IsAdvertising = isAdvertising;
        }
        
        /// <summary>
        /// Creates a new advertising state changed event.
        /// For events created on the Unity side.
        /// </summary>
        /// <param name="isAdvertising">Whether the device is currently advertising</param>
        public AdvertisingStateChangedEvent(bool isAdvertising)
            : base(BleHidEventType.AdvertisingStateChanged)
        {
            IsAdvertising = isAdvertising;
        }
        
        /// <summary>
        /// Gets a string representation of this event.
        /// </summary>
        public override string ToString()
        {
            return $"{base.ToString()}, isAdvertising={IsAdvertising}";
        }
    }
}
